generators/chipyard/src/main/scala/config/ChipConfigs.scala

```diff
@@ -34,5 +34,113 @@ class ChipLikeQuadRocketConfig extends Config(
   // Create the uncore clock group
   new chipyard.clocking.WithClockGroupsCombinedByName(("uncore", Seq("implicit", "sbus", "mbus", "cbus", "system_bus", "fbus", "pbus"), Nil)) ++
 
-  new chipyard.config.AbstractConfig)
+  new chipyard.config.AbstractConfig
+)
+
+/**
+  * Chip Config generated by the GUI
+  */
+class ExampleChipConfig extends Config(
+  // ==================================
```

Comment:
Can this be merged into ChipLikeRocketConfig?

Reply:
Where is the ChipLikeRocketConfig defined...?

Reply:
Actualy I forgot, hasn't been merged yet, after https://github.com/ucb-bar/chipyard/pull/1484 I change ChipLikeQuadRocket to ChipLikeRocket

Reply:
The ExampleChipConfig is generated by the GUI interface. It lists every Config fragment in the chip explicitly within this class. 

The idea is that this gives the user a clearer view of what components are included inside a chip, what Config fragment can be swapped with what else etc., and therefore can be used as a good teaching material for getting started tutorials (tutorials coming soon).

Listing Config fragments out in a single class also helps resolve the "what is overridden by what" problem. For example, for a tapeout chip, instead of overriding `WithXXXPunchthrough` with `WithXXXIOBinder`, the user will replace and switch between these two fragments for simulation vs. tapeout.

I'll try to merge this to the ChipLikeRocketConfig in future PR, but since this ExampleChipConfig is subject to user change (GUI will auto-generate the ChipConfigs.scala based on user configuration), and the scope can expand beyond using rocket cores, I suggest giving the class a more versatile name.

Reply:
At one point, `AbstractConfig` did not exist, and every single example config included every single relevant config fragment, as you describe. I fought for this model for quite some time, until it became untenable to maintain so many 50-line configs.

My perspective on "tapeout configs" is that they should be as explicit as possible, and will often require careful hand-edits/bespoke-configs. With that in mind, removing `AbstractConfig` from the ChipLike configs is a move I would support.

However, I'm not sure that using GUI-configs as "tapeout configs" is the right direction, given that the GUI will only be able to support a smaller design space than what the full system is capable of. 

I'd also prefer to consolidate any "ChipLike" example config into the `ChipLikeRocketConfig`, as currently there is CI/regressions around that Config that test many features, including a regression that builds and tests the FPGA bringup harness.

A proposal based on the above points:
- Expand `ChipLikeRocketConfig` to not override `AbstractConfig`. 
- Have GUI dump configs into a separate file. Since I expect users will often want to edit `ChipConfigs.scala` by hand, this also prevents edit conflicts. One idea... user specifies a name in the GUI... GUI dumps the file into `configs/GUI_<user_config_name>.scala`. 
- Make GUI flexible enough to support generating the exact same design (and exact same config text) as the `ChipLikeRocketConfig`.